core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -369,6 +369,23 @@ public static String hex(String value) {
     return Hex.encodeHexString(value.getBytes(UTF_8));
   }
 
+  /** SQL BIN(long) function. */
+  public static String bin(long value) {
+    int zeros = Long.numberOfLeadingZeros(value);
+    if (zeros == Long.SIZE) {
+      return "0";
+    } else {
+      int length = Long.SIZE - zeros;
+      byte[] bytes = new byte[length];
+      for (int index = length - 1; index >= 0; index--) {
+        bytes[index] = (byte) ((value & 0x1) == 1 ? '1' : '0');
```

Comment:
What does 0x1 stand for?

Reply:
Here stand for & with 1，if lowerest is 1 return 1，otherwise return 0.